core/mempool/src/lib.rs

```diff
@@ -142,18 +151,50 @@ impl Pool {
         Ok(())
     }
 
-    // TODO: Check snapshots on `pending_snapshots` which get unblocked and move them to `ready_snapshots`
-    pub fn add_payload(&self, payload: ChainPayload) -> Result<(), String> {
-        for transaction in payload.transactions {
-            self.add_transaction(transaction)?;
-        }
-        for receipt in payload.receipts {
-            self.add_receipt(receipt)?;
+    /// add requested transactions and receipts from peer. Move snapshot from pending_snapshots
```

Comment:
We should in general follow good punctuation and formatting in docstrings and error messages (see below). Start them with capital letters and when makes sense end with punctuation.